zjit/src/codegen.rs

```diff
@@ -730,9 +730,26 @@ fn gen_push_frame(asm: &mut Assembler, recv: Opnd) {
 fn param_reg(idx: usize) -> Reg {
     // To simplify the implementation, allocate a fixed register for each basic block argument for now.
     // TODO: Allow allocating arbitrary registers for basic block arguments
+    if idx >= ALLOC_REGS.len() {
+        unimplemented!(
+            "register spilling not yet implemented, too many basic block arguments ({}/{})",
+            idx + 1, ALLOC_REGS.len()
+        );
+    }
     ALLOC_REGS[idx]
 }
 
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    #[should_panic(expected = "register spilling not yet implemented, too many basic block arguments (11/10)")]
```

Comment:
The panic (by `unimplemented!`) itself works as an inline test, so I don't think you should test the test. It's also tedious to update the "11/10" part whenever you touch the `ALLOC_REGS` constant.